apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,1308 @@
+static const char *g_tlsname = "ws_test_tls_client";
+
+static int g_keep_alive_count = 0;
+static int g_chunk_count = 0;
+static int g_trailer = 0;
+static int g_normal_clen = 0;
+static int g_last_clen = 0;
+static int g_https = 0;
+static int g_tc_pass = 0;
+static int g_is_test_entity = 0;
+static const char HEADERFIELD_CONNECT[] = "Connection";
+static const char HEADERFIELD_CLOSE[] = "close";
+static const char HEADERFIELD_KEEPALIVE[] = "keep-alive";
+static const char HEADERFIELD_USERAGENT[] = "User-Agent";
+static const char HEADERFIELD_TINYARA[] = "TinyARA";
+
+
+static const char c_ca_crt_ec[] =
+	"-----BEGIN CERTIFICATE-----\r\n"
+	"MIICHzCCAaWgAwIBAgIBCTAKBggqhkjOPQQDAjA+MQswCQYDVQQGEwJOTDERMA8G\r\n"
+	"A1UEChMIUG9sYXJTU0wxHDAaBgNVBAMTE1BvbGFyc3NsIFRlc3QgRUMgQ0EwHhcN\r\n"
+	"MTMwOTI0MTU1MjA0WhcNMjMwOTIyMTU1MjA0WjA0MQswCQYDVQQGEwJOTDERMA8G\r\n"
+	"A1UEChMIUG9sYXJTU0wxEjAQBgNVBAMTCWxvY2FsaG9zdDBZMBMGByqGSM49AgEG\r\n"
+	"CCqGSM49AwEHA0IABDfMVtl2CR5acj7HWS3/IG7ufPkGkXTQrRS192giWWKSTuUA\r\n"
+	"2CMR/+ov0jRdXRa9iojCa3cNVc2KKg76Aci07f+jgZ0wgZowCQYDVR0TBAIwADAd\r\n"
+	"BgNVHQ4EFgQUUGGlj9QH2deCAQzlZX+MY0anE74wbgYDVR0jBGcwZYAUnW0gJEkB\r\n"
+	"PyvLeLUZvH4kydv7NnyhQqRAMD4xCzAJBgNVBAYTAk5MMREwDwYDVQQKEwhQb2xh\r\n"
+	"clNTTDEcMBoGA1UEAxMTUG9sYXJzc2wgVGVzdCBFQyBDQYIJAMFD4n5iQ8zoMAoG\r\n"
+	"CCqGSM49BAMCA2gAMGUCMQCaLFzXptui5WQN8LlO3ddh1hMxx6tzgLvT03MTVK2S\r\n"
+	"C12r0Lz3ri/moSEpNZWqPjkCMCE2f53GXcYLqyfyJR078c/xNSUU5+Xxl7VZ414V\r\n"
+	"fGa5kHvHARBPc8YAIVIqDvHH1Q==\r\n"
+	"-----END CERTIFICATE-----\r\n";
+
+static struct http_client_ssl_config_t g_ssl_config = {
+	(char *)c_ca_crt_ec, NULL, NULL,
+	sizeof(c_ca_crt_ec), 0, 0, WEBCLIENT_SSL_VERIFY_REQUIRED
+};
+
+//format
+static void ws_test_dump_usage(void)
+{
+	FUNC_EN;
+	printf("\n  webserver_test usage:\n");
+	printf("   $ webserver_test <iterations> <method> <uri> [options...] \n");
+	printf("\n");
+	printf(" <method>   : %%s (GET, PUT, POST, DELETE)\n");
+	printf(" <uri>      : %%s (Host address : should be started with http:// or https://)\n");
+	printf("\n [options...] \n");
+	printf(" test_entity=SIZE      Test entity size (default is 0)\n");
+	printf(" chunk-count=Y    	   Turns on Chunked-Transfer and sends Y chunks of test_entity (default Chunked-Transfer is OFF)\n");
+	printf(" keep-alive=X          Turns on keep-alive and sends X requests (default Keep-Alive is OFF)\n");
+	printf("\n  example:\n");
+	printf("   $ webserver_test 3 GET https://127.0.0.1/\n");
+	printf("   $ webserver_test 3 GET https://127.0.0.1/ test_entity=5\n");
+	FUNC_EX;
+}
+
+static char* ws_test_wget_strcpy(char *dest, const char *src, struct http_client_request_t *ws)
+{
+	FUNC_EN;
+	int len = strlen(src);
+
+	if (dest + len - ws->buffer >= ws->buflen) {
+		PRNT("Error: buffer is too small");
+		return NULL;
+	}
+
+	memcpy(dest, src, len);
+	dest[len] = '\0';
+	FUNC_EX;
+	return dest + len;
+}
+
+static char* ws_test_wget_strlencpy(char *dest, const char *src, int len, struct http_client_request_t *ws)
```

Comment:
better to move function like ws_test_wget_strlencpy, ws_test_wget_strcpy, gethostbyname etc to a webserver_test_utils file. it will make code easy to read and maintain
please share your opinion